### ISSUE TITLE
[Regression / 4.0.1] Inconsistency between 'org.eclipse.persistence.core/pom.xml' and 'org.eclipse.persistence.core/module-info.java'

### ISSUE THREAD

**Issue by jjohannes** [eclipse-ee4j/eclipselink#1883]
## Issue

When you add a dependency to `org.eclipse.persistence:org.eclipse.persistence.core:4.0.1` (in Gradle or Maven) and build/run a _Java Module_ you get this error:

```
Error occurred during initialization of boot layer 
ava.lang.module.FindException: Module org.eclipse.persistence.asm not found, required by org.eclipse.persistence.core
```

**This issue did not exist in `4.0.0`**

## Responsible change

The responsible change seems to be in #1849

```
-    requires static org.eclipse.persistence.asm;
+    requires org.eclipse.persistence.asm;
```

Which means that `org.eclipse.persistence.asm` is now always required at runtime:

https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/src/main/java/module-info.java#L29

However, the corresponding dependency remains `<optional>` and `<provided>` in the POM and is thus not automatically added to the runtime classpath:

https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/pom.xml#L47-L52

The module directive should either remain `compile static` or the corresponding POM entry should no longer be non-optional in _compile_ or _runtime_ scope.

**Comment by rfelcman** [comment]
Hello, I think, that this issue is fixed in 4.0.2 
https://github.com/eclipse-ee4j/eclipselink/releases/tag/4.0.2https://github.com/eclipse-ee4j/eclipselink/releases/tag/4.0.2
https://github.com/eclipse-ee4j/eclipselink/commit/b4dcd36ed94502663a8254c3c65cb9c01daafcbd#diff-d5a35c976cdd1cac92c245b23e17e7c678c1684604e14cd61b7ef084827a2365

Please leave message there if issue still remains.

**Comment by jjohannes** [comment]
Thanks. I can confirm that it is fixed in `4.0.2`.
